### PR TITLE
Release

### PR DESCRIPTION
**Pull request by PatrickDekker98**
updated some esp code,
added makefile.inc so it can be included like a library

### CODE REVIEW

File: code/esp_code/tcp_client.c [hunk 1]
```diff
@@ -81,6 +81,47 @@ static void tcp_client_task(void *pvParameters) {
...
+    // master is connected.
+    gpio_set_pull_mode(GPIO_MOSI, GPIO_PULLUP_ONLY);
+    gpio_set_pull_mode(GPIO_SCLK, GPIO_PULLUP_ONLY);
+    gpio_set_pull_mode(GPIO_CS, GPIO_PULLUP_ONLY);
+
+    // Initialize SPI slave interface
+    ret = spi_slave_initialize(HSPI_HOST, &buscfg, &slvcfg, 1);
+    assert(ret == ESP_OK);
+    //if (ret != ESP_OK) {
+     //   ESP_LOGE(TAG, "Failed to initialize spi slave");
```
Comment (JasperOosterbroek) [inline code/esp_code/tcp_client.c]: commented code should be removed

File: code/esp_code/tcp_client.c [hunk 2]
```diff
@@ -181,7 +185,7 @@ static void tcp_client_task(void *pvParameters) {
                 ESP_LOGI(TAG, "%s", sendbuf);
             }
 
-            // vTaskDelay(2000 / portTICK_PERIOD_MS);
+//            vTaskDelay(2000 / portTICK_PERIOD_MS);
```
Comment (JasperOosterbroek) [inline code/esp_code/tcp_client.c]: commented code should be removed

File: code/esp_code/tcp_client.c [hunk 3]
```diff
@@ -109,59 +150,22 @@ static void tcp_client_task(void *pvParameters) {
...
-
-        memset(recvbuf, 0, 129);
-        spi_slave_transaction_t t = {0};
-        memset(&t, 0, sizeof(t));
-
+        
         while (1) {
             // Clear receive buffer,
-            memset(recvbuf, 0x00, 129);
+            memset(recvbuf, 0xA5, 129);
```
Comment (JasperOosterbroek) [inline code/esp_code/tcp_client.c]: Is it documented why these values are used?

File: code/main.cpp [hunk 4]
```diff
@@ -7,15 +7,13 @@ int main(void) {
...
-    auto sclk = hwlib::target::pin_out(hwlib::target::pins::d9);
-    auto mosi = hwlib::target::pin_out(hwlib::target::pins::mosi);
-    auto miso = hwlib::target::pin_in(hwlib::target::pins::miso);
-    auto ss = hwlib::target::pin_out(hwlib::target::pins::d10);
+//auto ss = hwlib::target::pin_out(hwlib::target::pins::d4);
+    auto hand = hwlib::target::pin_in(hwlib::target::pins::d5);
 
     //configure spi bus with cs 0 = pin 10
     //configure spi bus with spi mode 1
     //configure with divider 42 ( 84 / 42 = 2Mhz )
```
Comment (JasperOosterbroek) [inline code/main.cpp:15]: shouldn't this value be changed?

File: code/src/esp_32.cpp [hunk 5]
```diff
@@ -2,14 +2,18 @@
...
+    esp_32_c::esp_32_c(hwlib::spi_bus &spi_connection, hwlib::pin_out &ss, hwlib::pin_in &hs)
+        : spi_connection(spi_connection),
+          slave_select(ss),
+          hand_shake(hs)
+          {}
 
     void esp_32_c::send(const r2d2::frame_external_s &frame) {
         // set the first bite of the buffer to 0
-        // only the first because teh fist byte of an external frame is its length
+        // only the first because the fist byte of an external frame is its length
```
Comment (JasperOosterbroek) [inline code/src/esp_32.cpp]: typo 

> fist

**Review by JasperOosterbroek — CHANGES_REQUESTED** [review]

**Review by gerritvanos — COMMENTED** [review]
I don't have more comments than jasper allready made if those are fixed i will approve asswel

File: code/src/esp_32.cpp (hunk 5 above)
Comment (PatrickDekker98) [reply code/src/esp_32.cpp]: typo where?

Comment (PatrickDekker98) [reply code/src/esp_32.cpp]: nvm

File: code/esp_code/tcp_client.c (hunk 3 above)
Comment (PatrickDekker98) [reply code/esp_code/tcp_client.c]: it was used for debugging

File: code/main.cpp (hunk 4 above)
Comment (PatrickDekker98) [reply code/main.cpp:15]: can you clarify on what you mean?

**Review by JasperOosterbroek — APPROVED** [review]

**Review by gerritvanos — APPROVED** [review]

**Pull request merged**